kedro/runner/parallel_runner.py

```diff
@@ -79,11 +79,11 @@ def save(self, data: Any):
             # Checks if the error is due to serialisation or not
             try:
                 pickle.dumps(data)
-            except Exception as exc:  # SKIP_IF_NO_SPARK
+            except Exception as inner_exc:  # SKIP_IF_NO_SPARK
                 raise DataSetError(
                     f"{str(data.__class__)} cannot be serialized. ParallelRunner "
                     "implicit memory datasets can only be used with serializable data"
-                ) from exc
+                ) from inner_exc
```

Comment:
L88 can just be `raise` then. I am not sure why we even have the `else` branch at all here, but that's maybe out of scope for this PR.